phoenix-core-client/src/main/java/org/apache/phoenix/schema/PMetaDataImpl.java

```diff
@@ -83,14 +83,27 @@ private void updateGlobalMetric(PTableRef pTableRef) {
 
     @Override
     public PTableRef getTableRef(PTableKey key) throws TableNotFoundException {
+        return getTableRef(key, false);
+    }
+
+    @Override
+    public PTableRef getTableRefOptimized(PTableKey key) {
+        try {
+            return getTableRef(key, true);
+        } catch (TableNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private PTableRef getTableRef(PTableKey key, boolean doNotThrowTableNotFoundException) throws TableNotFoundException {
```

Comment:
With this, you also mean to cover PHOENIX-7486?

Reply:
Actually, [PHOENIX-7486](https://issues.apache.org/jira/browse/PHOENIX-7486) is a more of generic JIRA for finding and fixing other occurrences of this anti pattern. I fixed one in here as that was contributing to the regressed numbers. W/o the change in `PMetadataImpl` I was able to bring down regression to 30-40% from 5K-4K% but need this change to bring it down to 0.

Reply:
nit: use `skipTNFE` instead of `doNotThrowTableNotFoundException`